genai-function-calling/openai-agents/main.py

```diff
@@ -0,0 +1,72 @@
+import asyncio
+import os
+
+from httpx import AsyncClient
+from agents import (
+    Agent,
+    ModelSettings,
+    OpenAIProvider,
+    RunConfig,
+    Runner,
+    function_tool,
+)
+from agents.tracing import GLOBAL_TRACE_PROVIDER
+from openai import AsyncAzureOpenAI
+
+# Shut down the global tracer as it sends to the OpenAI "/traces/ingest"
+# endpoint, which we aren't using and doesn't exist on alternative backends
+# like Ollama.
+GLOBAL_TRACE_PROVIDER.shutdown()
+
+
+@function_tool(strict_mode=False)
+async def get_latest_elasticsearch_version(major_version: int = 0) -> str:
+    """Returns the latest GA version of Elasticsearch in "X.Y.Z" format.
+
+    Args:
+        major_version: Major version to filter by (e.g. 7, 8). Defaults to latest
+    """
+    async with AsyncClient() as client:
```

Comment:
@anuraaga fyi interesting quirk is that if sync VCR doesn't record it. when I set everything async, it does. I didn't get the the bottom of why, but in any case doesn't hurt to make things async since that's the underlying client anyway